services/src/main/java/org/keycloak/forms/login/freemarker/FreeMarkerLoginFormsProvider.java

```diff
@@ -254,6 +254,10 @@ public Response createForm(String form) {
         Locale locale = session.getContext().resolveLocale(user);
         Properties messagesBundle = handleThemeResources(theme, locale);
 
+        if (authenticationSession != null && authenticationSession.getClientNote(Constants.KC_ACTION_EXECUTING) != null) {
```

Comment:
Will it makes sense to move this to method "createCommonAttributes" and remove the very same block from "createResponse" so that it is not same code duplicated on 2 places?

Reply:
Yes, it will be better. Thanks!